ofrak_core/test_ofrak/unit/test_ofrak_server.py

```diff
@@ -888,6 +888,40 @@ async def test_get_config(ofrak_client: TestClient, hello_world_elf):
     }
 
 
+async def test_search_string(ofrak_client, hello_world_elf):
+    create_resp = await ofrak_client.post(
+        "/create_root_resource", params={"name": "hello_world_elf"}, data=hello_world_elf
+    )
+    create_body = await create_resp.json()
+    resource_id = create_body["id"]
+    resp = await ofrak_client.post(
+        f"/{resource_id}/search_for_string",
+        json={"search_query": "hello", "case_ignore": True, "regex": True},
+    )
+    resp_body = await resp.json()
```

Comment:
We should probably add an assert that it actually finds the string. And we should make a similar check that it searches for one that isn't there and doesn't find it. 

Should be easy to get that from the body and assert.

That way we're actually confirming that this route works and does what it says it's supposed to.